includes/wc-update-functions.php

```diff
@@ -2175,3 +2175,69 @@ function wc_update_440_insert_attribute_terms_for_variable_products() {
 function wc_update_440_db_version() {
 	WC_Install::update_db_version( '4.4.0' );
 }
+
+/**
+ * Update DB version to 4.5.0.
+ */
+function wc_update_450_db_version() {
+	WC_Install::update_db_version( '4.5.0' );
+}
+
+/**
+ * Get coupons to update in the DB.
+ */
+function wc_update_450_get_coupons() {
+	global $wpdb;
+
+	$coupons = $wpdb->get_results( "SELECT ID, post_title FROM {$wpdb->posts} WHERE post_type = 'shop_coupon';", ARRAY_A );
+
+	update_option( 'woocommerce_update_350_coupons', $coupons );
+}
+
+/**
+ * Sanitize all coupons code.
+ *
+ * @return bool True to run again, false if completed.
+ */
+function wc_update_450_sanitize_coupons_code() {
+	global $wpdb;
+
+	$coupons = array_filter( (array) get_option( 'woocommerce_update_350_coupons', array() ) );
```

Comment:
What do you think about changing the algorithm a little bit:
1. Instead of the post title and ID of all coupons, store only one ID starting from zero.
2. Then in each iteration, fetch using `where id > $stored_id limit 10`, and process them. Update the ID to last ID.
3. Stop when there are no more Ids.

This way we won't have to do any large inserts and updates, and we will prevent the race condition that I mentioned in above comment.